### PR TITLE
[buffer orch] Bugfix: Don't query counter SAI_BUFFER_POOL_STAT_XOFF_ROOM_WATERMARK_BYTES on a pool where it is not supported

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
Currently, SAI_BUFFER_POOL_STAT_WATERMARK_BYTES and SAI_BUFFER_POOL_STAT_XOFF_ROOM_WATERMARK_BYTES are queried for buffer pools.
However, the latter is not supported on all pools and all platforms, which will result in sairedis complaint like this:
```
Jun 17 23:31:34.949613 sonic ERR syncd#SDK: :- updateSupportedBufferPoolCounters: BUFFER_POOL_WATERMARK_STAT_COUNTER: counter SAI_BUFFER_POOL_STAT_XOFF_ROOM_WATERMARK_BYTES is not supported on buffer pool oid:0x1000000018, rv: SAI_STATUS_NOT_IMPLEMENTED
Jun 17 23:31:34.951915 sonic ERR syncd#SDK: :- updateSupportedBufferPoolCounters: BUFFER_POOL_WATERMARK_STAT_COUNTER: counter SAI_BUFFER_POOL_STAT_XOFF_ROOM_WATERMARK_BYTES is not supported on buffer pool oid:0x1200000018, rv: SAI_STATUS_NOT_IMPLEMENTED
Jun 17 23:31:34.954177 sonic ERR syncd#SDK: :- updateSupportedBufferPoolCounters: BUFFER_POOL_WATERMARK_STAT_COUNTER: counter SAI_BUFFER_POOL_STAT_XOFF_ROOM_WATERMARK_BYTES is not supported on buffer pool oid:0x400000018, rv: SAI_STATUS_NOT_IMPLEMENTED
```
To avoid that, we need to test whether it is supported before putting it to FLEX_COUNTER_DB

Signed-off-by: Stephen Sun <stephens@nvidia.com>

**Why I did it**

**How I verified it**
Run vstest and manually test.

**Details if related**

### CODE REVIEW

**Comment by prsunny** [comment]
@neethajohn , could you please the code? Also, I dont think we can backport to 202012 as this might break other platforms implementations. We already had an issue backporting this PR - https://github.com/Azure/sonic-swss/pull/1789. Hence, removing label.

**Review by neethajohn — APPROVED** [review]

File: orchagent/bufferorch.cpp [hunk 1]
```diff
@@ -218,29 +228,60 @@ void BufferOrch::generateBufferPoolWatermarkCounterIdList(void)
...
     {
-        statList += (sai_serialize_buffer_pool_stat(it) + list_item_delimiter);
+        statListPoolOnly += (sai_serialize_buffer_pool_stat(it) + list_item_delimiter);
     }
-    if (!statList.empty())
+    string statListPoolAndShp = statListPoolOnly;
+    for (const auto &it : bufferSharedHeadroomPoolWatermarkStatIds)
     {
-        statList.pop_back();
+        statListPoolAndShp += (sai_serialize_buffer_pool_stat(it) + list_item_delimiter);
```
Comment (qiluo-msft) [inline orchagent/bufferorch.cpp:239]: >statListPoolAndShp [](http://example.com/codeflow?start=8&length=18)

It is difficult to read `Shp`. Could you add some comment?

Comment (stephenxs) [reply orchagent/bufferorch.cpp:239]: How about the name `statListBufferPoolAndSharedHeadroomPool` ?

Comment (qiluo-msft) [reply orchagent/bufferorch.cpp:239]: Yes, I can read now.

Comment (stephenxs) [reply orchagent/bufferorch.cpp]: Fixed.

**Review by qiluo-msft — APPROVED** [review]
LGTM

**Review by neethajohn — APPROVED** [review]

**Pull request merged**